tracer/src/Datadog.Trace/AppSec/ObjectExtractor.cs

```diff
@@ -56,13 +57,22 @@ internal static class ObjectExtractor
 
         private static IReadOnlyDictionary<string, object?> ExtractProperties(object body, int depth, HashSet<object> visited)
         {
-            if (visited.Contains(body))
+            try
             {
+                if (visited.Contains(body))
+                {
+                    return EmptyDictionary;
+                }
+
+                visited.Add(body);
+            }
+            catch
+            {
+                // Contains and Add call GetHashCode which can throw an exception if has a custom implementation
+                // If visited is empty, we could potentially get the exception only when calling Add
```

Comment:
So this is client's implementation of `.GetHashCode()` that is throwing the exception? I wonder if there are other places where we handle client objects and this could occur?